### ISSUE TITLE
Storing a reference to Date.now breaks fake timer based tests

### ISSUE THREAD

**Issue by timotheeg** [moment/moment#2855]
The upgrade to ```2.11.0``` breaks builds using fake timers (e.g. [sinon's fake timers](http://sinonjs.org/docs/#clock)). Because the new code stores a reference to ```Date.now``` instead of doing a dynamic access.

Sample test case:
```javascript
var
	sinon = require('sinon'),
	moment = require('moment');

sinon.clock = sinon.useFakeTimers();

console.log((new Date()).toString());
console.log(moment().toString());
```

Output with moment 2.10.6:
```bash
Thu Jan 01 1970 08:00:00 GMT+0800 (SGT)
Thu Jan 01 1970 08:00:00 GMT+0800
```

Output with moment 2.11.0:
```bash
Thu Jan 01 1970 08:00:00 GMT+0800 (SGT)
Mon Jan 04 2016 12:52:13 GMT+0800
```

From a npm perspective, if a project uses ```moment-timezone```, then because the dependency on moment is ```>= 2.6.0```, you get an automatic upgrade to ```2.11.0``` and travis starts faiing on the fake timers.

The fix is simple: use moment's hook ```moment.now``` for tests instead, so this issue can be solved with a simple:
```javascript
moment.now = function() { return Date.now(); }
```

I'm opening the issue anyway because storing references to internal functions is a bit of a dodgy practice, and for reference in case anybody else using ```sinon``` and ```moment``` start seeing failing builds.

**Comment by nowells** [comment]
There is a PR I created to solve this: https://github.com/moment/moment/pull/2858

**Comment by nowells** [comment]
Sorry I missed this issue before creating https://github.com/moment/moment/issues/2857! But the PR to fix should hopefully get this resolved quickly.

**Comment by timotheeg** [comment]
Thanks a lot @nowells . I've patched my project builds to use the internal ```moment.now``` hook, but I do think restoring the ability to use fake timers transparently is a very good thing. Thanks for making a PR!

**Comment by mj1856** [comment]
Closed in favor of PR #2858.  Thanks.
